src/parsers.rs

```diff
@@ -0,0 +1,208 @@
+        e.inductor_output_code.as_ref().map(|m| Metadata::InductorOutputCode(m))
+    }
+
+    fn parse<'e>(&self,
+        lineno: usize,
+        metadata: Metadata<'e>,
+        _rank: Option<u32>,
+        compile_id: &Option<CompileId>,
+        payload: &str
+    ) -> anyhow::Result<ParseOutput> {
+        if let Metadata::InductorOutputCode(metadata) = metadata {
+            let filename = metadata
+                .filename
+                .as_ref()
+                .and_then(|p| Path::file_stem(p))
+                .map_or_else(
+                    || PathBuf::from("inductor_output_code.txt"),
+                    |stem| {
+                        let mut r = OsString::from("inductor_output_code_");
+                        r.push(stem);
+                        r.push(OsStr::new(".txt"));
+                        r.into()
+                    },
+            );
+            simple_file_output(&filename.to_string_lossy(), lineno, compile_id, payload)
+        } else {
+            Err(anyhow::anyhow!("Expected InductorOutputCode metadata"))
+        }
+    }
+}
+
+pub struct OptimizeDdpSplitChildParser;
+impl StructuredLogParser for OptimizeDdpSplitChildParser {
+    fn name(&self) -> &'static str {
+        "optimize_ddp_split_child"
+    }
+    fn get_metadata<'e>(&self, e: &'e Envelope) -> Option<Metadata<'e>> {
+        e.optimize_ddp_split_child.as_ref().map(|m| Metadata::OptimizeDdpSplitChild(m))
+    }
+
+    fn parse<'e>(&self,
+        lineno: usize,
+        metadata: Metadata<'e>,
+        _rank: Option<u32>,
+        compile_id: &Option<CompileId>,
+        payload: &str
+    ) -> anyhow::Result<ParseOutput> {
+        if let Metadata::OptimizeDdpSplitChild(m) = metadata {
+            let filename = format!("optimize_ddp_split_child_{}.txt", m.name);
+            simple_file_output(&filename, lineno, compile_id, payload)
+        } else {
+            Err(anyhow::anyhow!("Expected OptimizeDdpSplitChild metadata"))
+        }
+    }
+}
+
+// Register your parser here
+pub fn all_parsers<'t>(tt: &'t TinyTemplate<'t>) -> Vec<Box<dyn StructuredLogParser + 't>> {
+    // We need to use Box wrappers here because vecs in Rust need to have known size
+    let result : Vec<Box<dyn StructuredLogParser>> = vec![
+        Box::new(SentinelFileParser::new("optimize_ddp_split_graph", |e| e.optimize_ddp_split_graph.as_ref())),
+        Box::new(SentinelFileParser::new("compiled_autograd_graph", |e| e.compiled_autograd_graph.as_ref())),
+        Box::new(SentinelFileParser::new("aot_forward_graph", |e| e.aot_forward_graph.as_ref())),
+        Box::new(SentinelFileParser::new("aot_backward_graph", |e| e.aot_backward_graph.as_ref())),
+        Box::new(SentinelFileParser::new("aot_joint_graph", |e| e.aot_joint_graph.as_ref())),
+        Box::new(SentinelFileParser::new("inductor_post_grad_graph", |e| e.inductor_post_grad_graph.as_ref())),
+        Box::new(DynamoOutputGraphParser),
+        Box::new(DynamoGuardParser { tt }),
```

Comment:
I wonder if the individual analyses should be responsible for their own template instances themselves